### DISCUSSION TITLE
[IntelliSense] allow highlighting of classes in constants or maps

### DISCUSSION THREAD

**Discussion by phoebejaffe** [tailwindlabs/tailwindcss#12356] [Ideas]
Our codebase relies heavily on className strings that come from constants or maps, rather than living in the JSX directly. This allows us to override classes based on some state, or an override file passed down via React Context. 

Here's a simplified example:
```
// classOverrides.ts
const overrides = {
  MyJsxFile: {
    root: 'text-14 font-bold'
  }
}

// MyJsxFile.tsx
const overrides = useClassOverrides('MyJsxFile');

<div className={overrides.root ?? 'text-16'}>
  /* ... */
</div>
```
This could work by specifying that highlighting should happen in strings inside any file matching a glob, but I'm sure there are many good ways to design this.

This feature request also applies to situations like the following:
```
const userClassName = userActive ? '' : 'opacity-50';
```
This could work by specifying in config that all string constants whose names end with `ClassName` should get Tailwinds Intellisense applied.

It'd be SO amazing to get tailwinds IntelliSense in our overrides files, since we do a lot of work in there. From looking at the config, it doesn't seem there's currently a way to specify extra places that we want the IntelliSense behavior. Is this something y'all would consider adding?

**Comment by wongjn** [comment]
You could try configuring the [`tailwindCSS.experimental.classRegex` setting](https://github.com/tailwindlabs/tailwindcss/issues/7553#issuecomment-735915659).

**Comment by phoebejaffe** [comment]
Bah, it was there all along. Sorry about that, I somehow missed it while searching around.

Thank you for an amazing plugin and project, btw! <3

**Comment by phoebejaffe** [comment]
Wait, @wongjn is there a reason this isn't documented anywhere?

**Reply by wongjn** [reply]
I'd assume, as the name suggests, it is an experimental feature.

**Reply by phoebejaffe** [reply]
ah, sorry, assumed it was your choice. But yeah, the feature seems to work, and it's been around since 2020... just cause it's considered experimental doesn't mean it's not worth documenting IMO. At least speaking for myself, I've looked for this at least twice before over the last year, and I only found it by opening an issue. 😆
